pandas/core/config_init.py

```diff
@@ -279,7 +280,8 @@ def mpl_style_cb(key):
 
 with cf.config_prefix('display'):
     cf.register_option('precision', 6, pc_precision_doc, validator=is_int)
-    cf.register_option('float_format', None, float_format_doc)
+    cf.register_option('float_format', None, float_format_doc,
+                       validator=is_callable)
```

Comment:
I *think* you can do:

``validator=is_instance_factory([type(None), is_callable)``